### DISCUSSION TITLE
HL7 Parsing

### DISCUSSION THREAD

**Discussion by anagen** [senaranya/HL7#81] [Q&A] [answered]
I have a need to parse a HL7 data file and map data to a Database table
DB integration is probably easy, just wanted to understand how  parsing would work using PHP 

I am new to PHP, have setup the PHP environment on Ubuntu, can someone guide me how 
do I leverage this Package / Library please ?

Thanks in advance

Nagen

**Comment by senaranya** [comment, marked as answer]
* Install [composer](https://getcomposer.org/download/) unless already installed
* Create a directory for your project, along with a directory `src` inside it
* Inside the project directory, run the command: `composer require aranyasen/hl7` (this installs the library in vendor/ directory)
* Add a basic script, say `src/hl7_demo.php`:
```php
<?php

declare(strict_types=1);

namespace src;

require_once 'vendor/autoload.php';

use Aranyasen\HL7\Message;

$msg = new Message("MSH|^~\\&|1|\rPV1|1|O|^AAAA1^^^BB|", null, true); // Create Message object from a HL7 string
echo $msg->toString(true); // Print it to make sure the object was created successfully
```
* use `php src/hl7_demo.php` to run it

Note: This is a bare-minimum script to demonstrate how to use this library. For details on functionalities, check out the [documentation](https://github.com/senaranya/HL7/blob/master/README.md)
